recipes/gmp/all/conanfile.py

```diff
@@ -36,46 +42,79 @@ def package_id(self):
         del self.info.options.run_checks  # run_checks doesn't affect package's ID
 
     def build_requirements(self):
-        if tools.os_info.is_windows and self.settings.compiler != "Visual Studio" and \
-           "CONAN_BASH_PATH" not in os.environ and tools.os_info.detect_windows_subsystem() != "msys2":
+        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
             self.build_requires("msys2/20200517")
+        if self.settings.compiler == "Visual Studio":
+            self.build_requires("yasm/1.3.0")
```

Comment:
Why only for Visual Studio? Is it not also required for all platforms if `disable_assembly=False`?

Reply:
Other platforms can just use the gnu assembler.

Reply:
By using yasm, this also means that for the moment only x86 and x86_64 architectures are supported on MSVC.
Maybe it is possible to combine the GNU assembler (from binutils) with MSVC, but that's something I did not test.

Reply:
Ok, so LGTM